src/utils/presets/migrateTreasuryFunds.js

```diff
@@ -0,0 +1,123 @@
+import { ethers } from 'ethers'
+import { erc20ABI } from 'wagmi'
+import { readContract } from '@wagmi/core'
+
+import { getEthPNTAsset, getRawAmount, pNetworkV2Vault } from './utils'
+import settings from '../../settings'
+
+const migrateTreasuryFunds = ({ presetParams, setPresetParams }) => ({
+  id: 'migrateTreasuryFunds',
+  name: 'Migrate Treasury Funds',
+  description: 'Pegin Treasury funds to new chain',
+  args: [
+    {
+      id: 'select-token-address',
+      name: 'tokenAddress',
+      component: 'AssetSelection',
+      props: {
+        onSelect: (_address) =>
+          setPresetParams({
+            ...presetParams,
+            0: _address
+          })
+      }
+    },
+    {
+      id: 'input-amount',
+      name: 'amount',
+      component: 'Input',
+      props: {
+        type: 'number',
+        style: {
+          fontSize: 15
+        },
+        placeholder: 'Amount ...',
+        value: presetParams[1] || '',
+        onChange: (_e) =>
+          setPresetParams({
+            ...presetParams,
+            1: _e.target.value
+          })
+      }
+    },
+    {
+      id: 'select-chain',
+      name: 'destinationChain',
+      component: 'ChainSelection',
+      props: {
+        onSelect: (_chainId) =>
+          setPresetParams({
+            ...presetParams,
+            2: _chainId
+          })
+      }
+    },
+    {
+      id: 'input-receiver-address',
+      name: 'receiverAddress',
+      component: 'Input',
+      props: {
+        style: {
+          fontSize: 15
+        },
+        placeholder: 'New Treasury Address (receiver address) ...',
+        value: presetParams[3] || '',
+        onChange: (_e) =>
+          setPresetParams({
+            ...presetParams,
+            3: _e.target.value
+          })
+      }
+    },
+    {
+      id: 'input-user-data',
+      name: 'userData',
+      component: 'Input',
+      props: {
+        style: {
+          fontSize: 15
+        },
+        placeholder: 'Optional User Data ...',
+        value: presetParams[4] || '',
+        onChange: (_e) =>
+          setPresetParams({
+            ...presetParams,
+            4: _e.target.value
+          })
+      }
+    }
+  ],
+  prepare: async () => {
```

Comment:
This can't work! The script is executed by dandelion voting contract:
you are just doing a peg-in here; the contract does not hold funds... you need to first transfer from the vault to the dandelion contract, and also approve erc20 vault to spend the transferred tokens.

Reply:
fixed